### PR TITLE
fix/invalid exam score

### LINKED ISSUE #125: [BUG] Exam mode score issue
**Describe the bug**
Some users reported the in the exam mode the final score is never 100 even if they answer correctly all the questions.
https://github.com/Ditectrev/Practice-Exams-Platform/pull/45#issuecomment-2342986040

**To Reproduce**
1. Go to Exam Mode
2. Answer all questions correctly
4. See score

**Expected behavior**
If all questions are correctly answered the score should be 100.

**Screenshots**
image

### PR DESCRIPTION
**Pull request by chaitanya-bhargava**
Fixes #125 

- [ x ] I have read and agree to the [Code of Conduct](https://github.com/eduardconstantin/azure-fundamentals/blob/main/CODE_OF_CONDUCT.md), and have followed the [Contributing guidelines](https://github.com/eduardconstantin/azure-fundamentals/blob/main/CONTRIBUTING.md).

## Pull Request Type

- [ x ] Bug fix (please link related issues if applicable)

## Summary
Before:
The exam score was incorrect as the last answer was not being counted during quiz form submission. This was happening due to of order asynchronous execution of the saveAnswers and reCount utilities. 

After:
I have separated the quiz form submit logic from the nextQuestion handler and made it depend on changes in savedAnswers so that the quiz form gets submitted only when there are no null entries in the savedAnswers array.

Please review the PR! Thank you!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
   },
   "dependencies": {
     "@apollo/client": "^3.7.9",
+    "@apollo/server": "^4.11.0",
```
Comment (eduardconstantin) [inline package.json:19]: Do we need apollo server?
```suggestion
```

**Review by eduardconstantin — CHANGES_REQUESTED** [review]
I tested on a few questions and it counts the correct score now.
Just one request regarding the apollo server package.

File: package.json (hunk 1 above)
Comment (chaitanya-bhargava) [reply package.json:19]: I encountered an error saying that the app could not resolve @apollo/server when I began the exam. Installing it fixed the error

Comment (chaitanya-bhargava) [reply package.json:19]: @eduardconstantin @apollo/server package is being used [here](https://github.com/eduardconstantin/azure-cloud-exams/blob/main/app/api/graphql/route.ts) in line 3

Comment (eduardconstantin) [reply package.json:19]: That's strange, the deployed app is working fine without it

**Review by eduardconstantin — APPROVED** [review]
Let's keep the package, I'll check it later this week.
Thank you for your contribution!

**Pull request merged**